src/order/sublattice.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2020 Matúš Behun. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matúš Behun
+-/
+import order.lattice
+
+/-!
+    Definition of sublattice based on typeclass of lattice
+-/
+
+variables { L : Type* } [ lattice L ] { s : set L }
+
+structure sublattice ( L : Type* ) [ lattice L ] :=
+  ( carrier : set L)
+  ( inf_mem'  {a b : L} : a ∈ carrier → b ∈ carrier → a ⊓ b ∈ carrier )
+  ( sup_mem'  {a b : L} : a ∈ carrier → b ∈ carrier → a ⊔ b ∈ carrier )
+
+namespace sublattice
+
+instance : has_coe (sublattice L) (set L) := ⟨sublattice.carrier⟩
+
+instance : has_mem L (sublattice L) := ⟨ λ m S, m ∈ ( S : set L) ⟩
+
+instance : has_coe_to_sort (set L) := ⟨_, λ s, {x // x ∈ s}⟩
```

Comment:
This was the right way of doing things until a week or so ago, but now you should replace these three instances with `set_like`. Have a look at `submonoid` for an example.

Reply:
Solved in commit cd29f8e